src/main/fc/config.c

```diff
@@ -284,7 +285,7 @@ static void validateAndFixConfig(void)
 #endif
 
     if (
-        featureIsEnabled(FEATURE_3D) || !featureIsEnabled(FEATURE_GPS)
+        featureIsEnabled(FEATURE_3D) || (!featureIsEnabled(FEATURE_GPS) && gpsConfig()->provider != GPS_MSP)
```

Comment:
Ok, this is slightly different to how I thought this should be done. I thought that even if `gpsConfig()->provider == GPS_MSP`, `featureIsEnabled(FEATURE_GPS)` should be required for GPS data over MSP to be used - this seems to be a more consistent usage. This will not require the `gpsConfig()->provider != GPS_MSP` check here.

Reply:
OK